jni/tools/native_loader/src/main/java/com/github/fmeum/rules_jni/RulesJni.java

```diff
@@ -182,14 +182,29 @@ private static Path extractLibrary(String basename, URL libraryResource) throws
     int lastDot = mappedName.lastIndexOf('.');
     Path tempFile = Files.createTempFile(
         tempDir, mappedName.substring(0, lastDot) + "_", mappedName.substring(lastDot));
+
     try (InputStream in = libraryResource.openStream()) {
       Files.copy(in, tempFile, StandardCopyOption.REPLACE_EXISTING);
     }
     return tempFile;
   }
 
+  private static boolean isAndroid(){
+    String javaVendor = System.getProperty("java.vendor", "");
+    return javaVendor.contains("Android");
+  }
+
   private static Path getOrCreateTempDir() throws IOException {
     if (tempDir == null) {
+      if (RulesJni.isAndroid()){
+        String androidTempPath = "/data/local/tmp/rules_jni";
```

Comment:
getCacheDir should give "/data/data/<your_app>/cache", and "java.tmp.dir" is null and java.io/tmpdir is "/tmp" (which is incorrect).

if this path doesn't exist, then we will have access to create it anyways and it will be created at mkdir